rdr_service/offline/study_nph_biobank_file_export.py

```diff
@@ -0,0 +1,254 @@
+        ordered_sample_ids = [sample_update.rdr_ordered_sample_id for sample_update in sample_updates]
+        ordered_samples: Iterable[OrderedSample] = session.query(OrderedSample).filter(
+            OrderedSample.id.in_(ordered_sample_ids)
+        ).distinct().all()
+
+    nph_orders_dao = NphOrderDao()
+    with nph_orders_dao.session() as session:
+        order_ids = list(ordered_sample.order_id for ordered_sample in ordered_samples)
+        orders: Iterable[Order] = session.query(Order).filter(
+            Order.id.in_(order_ids)
+        ).distinct().all()
+
+    return orders
+
+
+@lru_cache(maxsize=128, typed=False)
+def _get_study_category(study_category_id: int) -> StudyCategory:
+    study_category_dao = NphStudyCategoryDao()
+    with study_category_dao.session() as session:
+        return session.query(StudyCategory).get(study_category_id)
+
+
+@lru_cache(maxsize=128, typed=False)
+def _get_parent_study_category(study_category_id: int) -> StudyCategory:
+    study_category = _get_study_category(study_category_id)
+    return _get_study_category(study_category.parent_id)
+
+
+def _get_ordered_samples(order_id: int) -> List[OrderedSample]:
+    ordered_sample_dao = NphOrderedSampleDao()
+    with ordered_sample_dao.session() as session:
+        return session.query(OrderedSample).filter(
+            and_(
+                OrderedSample.order_id == order_id,
+                OrderedSample.aliquot_id.is_not(None)
+            )
+        ).all()
+
+
+def _convert_ordered_samples_to_samples(order_id: int, ordered_samples: List[OrderedSample]) -> List[Dict[str, Any]]:
+    samples = []
+    for ordered_sample in ordered_samples:
+        supplemental_fields = ordered_sample.supplemental_fields if ordered_sample.supplemental_fields else {}
+        notes = ", ".join([f"{key}: {value}" for key, value in supplemental_fields.items()])
+        extract_volume_units = findall("[a-zA-Z]+", ordered_sample.volume)
+        volume_units = extract_volume_units[-1] if extract_volume_units else ""
+        sample = {
+            "sampleID": ordered_sample.aliquot_id,
+            "specimenCode": ordered_sample.identifier,
+            "kitID": order_id if ordered_sample.identifier.startswith("ST") else "",
+            "volume": ordered_sample.volume,
+            "volumeUOM": volume_units,
+            "processingDateUTC": ordered_sample.finalized,
+            "notes": notes,
+        }
+        samples.append(sample)
+    return samples
+
+
+def _get_rdr_participant_summary_for_nph_partipant(nph_participant_id: int) -> Optional[RdrParticipantSummary]:
+    rex_participant_mapping_dao = RexParticipantMappingDao()
+    with rex_participant_mapping_dao.session() as rex_sm_session:
+        rex_participant_mapping: RexParticipantMapping = (
+            rex_sm_session.query(RexParticipantMapping).filter(
+                RexParticipantMapping.ancillary_participant_id == nph_participant_id
```

Comment:
Can you add a filter here for `RexParticipantMapping.ancillary_study_id == 2`? Sooner than later we will have more studies.